crates/deltalake-core/src/operations/create.rs

```diff
@@ -181,8 +181,11 @@ impl CreateBuilder {
     ///
     /// This might include provenance information such as an id of the
     /// user that made the commit or the program that created it.
-    pub fn with_metadata(mut self, metadata: Map<String, Value>) -> Self {
-        self.metadata = Some(metadata);
+    pub fn with_metadata(
+        mut self,
+        metadata: impl IntoIterator<Item = (String, serde_json::Value)>,
```

Comment:
This looks interesting, would love to understand the reason for changing. Is this to allow more than just Map?

Reply:
Mostly to streamline between the other implementations, we were using HashMap there, also the commit takes a hashmap as input and not a map